webpack/assets/javascripts/react_app/components/SettingRecords/SettingRecordsSelectors.js

```diff
@@ -0,0 +1,10 @@
+import { propsToCamelCase } from '../../common/helpers';
+
+const selectSettingRecords = state => state.settingRecords;
+export const selectSettings = state => selectSettingRecords(state).settings;
+export const selectSettingsByCategory = (state, category) =>
+  selectSettings(state)[category].map(propsToCamelCase);
+export const selectSettingById = (state, id, category) =>
+  selectSettingsByCategory(state, category).find(setting => setting.id === id);
```

Comment:
Where using `map/find` in a selector it usually makes sense to memorize them to avoid unnecessary calculations.